src/core/Merger.cpp

```diff
@@ -624,5 +621,18 @@ Merger::ChangeList Merger::mergeMetadata(const MergeContext& context)
             changes << tr("Adding missing icon %1").arg(QString::fromLatin1(customIconId.toRfc4122().toHex()));
         }
     }
+
+    // CustomData is merged with entries of the new customData overwrite entries
+    // of the older CustomData - the dict with the newest entry is considered
```

Comment:
@varjolintu not sure how relevant that comment is. We aren't considering the date at all when overwriting.

I think we should give priority to the custom data in the target database if it exists, and only `set` if it's not already in there. This is pretty much what we do with the custom icons just above.

Reply:
Maybe that is the best, yes. I'll make the modifications.